javatests/arcs/android/devtools/DevToolsStarterTest.kt

```diff
@@ -0,0 +1,112 @@
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.android.devtools
+
+import android.app.Application
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import arcs.android.devtools.DevToolsService.Companion.STORAGE_CLASS
+import arcs.sdk.android.storage.service.StorageService
+import com.google.common.truth.Truth.assertThat
+import kotlinx.coroutines.runBlocking
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.Robolectric
+import org.robolectric.Shadows.shadowOf
+import org.robolectric.shadows.ShadowService
+
+@RunWith(AndroidJUnit4::class)
+class DevToolsStarterTest {
+
+  private lateinit var app: Application
+  private lateinit var context: Context
+  private lateinit var devToolsIntent: Intent
+
+  @Before
+  fun setUp() {
+    context = ApplicationProvider.getApplicationContext<Application>()
+    devToolsIntent = Intent(context, DevToolsService::class.java)
+    app = ApplicationProvider.getApplicationContext()
+  }
+
+  @Test
+  fun start_createsIntent() = lifecycle { _, serviceShadow ->
+    val devToolsStarter = DevToolsStarter(context)
+    val bundle = Bundle().apply {
+      putSerializable(STORAGE_CLASS, StorageService::class.java)
+    }
+    devToolsIntent.putExtras(bundle)
+
+    devToolsStarter.start()
+    val actualIntent = serviceShadow.nextStartedService
+
+    assertThat(actualIntent.filterEquals(devToolsIntent)).isTrue()
+    assertThat(actualIntent.extras?.getSerializable(STORAGE_CLASS))
+      .isEqualTo(devToolsIntent.extras?.getSerializable(STORAGE_CLASS))
+  }
+
+  @Test
+  fun start_withStorageService_appliesBundle() = lifecycle { _, serviceShadow ->
+      val devToolsStarter = DevToolsStarter(context)
+      val bundle = Bundle().apply {
+        putSerializable(STORAGE_CLASS, MySecondStorageService::class.java)
+      }
+      devToolsIntent.putExtras(bundle)
+
+      devToolsStarter.start(MySecondStorageService::class.java)
+      val actualIntent = serviceShadow.nextStartedService
+
+      assertThat(actualIntent.filterEquals(devToolsIntent)).isTrue()
+      assertThat(actualIntent.extras?.getSerializable(STORAGE_CLASS))
+        .isEqualTo(devToolsIntent.extras?.getSerializable(STORAGE_CLASS))
+    }
+
+  @Test
+  fun start_withBadStorageService_createsIntent() = lifecycle { _, serviceShadow ->
+    val devToolsStarter = DevToolsStarter(context)
+
+    devToolsStarter.start(MyClass::class.java)
+    val actualIntent = serviceShadow.nextStartedService
+
+    assertThat(actualIntent.filterEquals(devToolsIntent)).isTrue()
+    assertThat(actualIntent.extras?.getSerializable(STORAGE_CLASS))
+      .isEqualTo(devToolsIntent.extras?.getSerializable(STORAGE_CLASS))
+  }
```

Comment:
This is a great test! It makes me think we may want to throw an exception if the user passes a non-StorageService subclass instead of silently using StorageService if what they pass is invalid. (Better to crash on unexpected input than give unexpected behavior)